reactor-core/src/main/java/reactor/core/publisher/FluxBuffer.java

```diff
@@ -153,9 +153,9 @@ public void onNext(T t) {
 				buffer = b;
 			}
 
-			b.add(t);
-
-			if (b.size() == size) {
+			if (!b.add(t)) {
```

Comment:
The item would need to be discarded properly if it fails to be added. https://projectreactor.io/docs/core/release/reference/#cleanup

Reply:
Fixed!